librz/core/canalysis.c

```diff
@@ -2311,61 +2311,124 @@ RZ_API int rz_core_print_bb_gml(RzCore *core, RzAnalysisFunction *fcn) {
 	return true;
 }
 
-RZ_API void rz_core_analysis_datarefs(RzCore *core, ut64 addr) {
-	RzAnalysisFunction *fcn = rz_analysis_get_fcn_in(core->analysis, addr, -1);
-	if (fcn) {
-		bool found = false;
-		const char *me = fcn->name;
-		RzListIter *iter;
-		RzAnalysisXRef *xref;
-		RzList *xrefs = rz_analysis_function_get_xrefs_from(fcn);
-		rz_list_foreach (xrefs, iter, xref) {
-			RzBinObject *obj = rz_bin_cur_object(core->bin);
-			RzBinSection *binsec = rz_bin_get_section_at(obj, xref->to, true);
-			if (binsec && binsec->is_data) {
-				if (!found) {
-					rz_cons_printf("agn %s\n", me);
-					found = true;
-				}
-				RzFlagItem *item = rz_flag_get_i(core->flags, xref->to);
-				const char *dst = item ? item->name : sdb_fmt("0x%08" PFMT64x, xref->to);
-				rz_cons_printf("agn %s\n", dst);
-				rz_cons_printf("age %s %s\n", me, dst);
+static inline bool is_between(ut64 a, ut64 x, ut64 b) {
+	return (a == UT64_MAX && b == UT64_MAX) || RZ_BETWEEN(a, x, b);
+}
+
+static inline char *core_flag_name(const RzCore *core, ut64 addr) {
+	RzFlagItem *item = rz_flag_get_i(core->flags, addr);
+	return item ? strdup(item->name) : rz_str_newf("0x%08" PFMT64x, addr);
+}
+
+static inline void core_graph_dataref(RzCore *core, RzAnalysisFunction *fcn, RzGraph *graph) {
+	if (!fcn) {
+		RZ_LOG_INFO("Not in a function. Use 'df' to define it.\n");
```

Comment:
I'd remove this... `df` works in visual mode, but not elsewhere. Also, command names could change in the future.